docs/dev/opensearch-same-table-join-query.md

```diff
@@ -0,0 +1,1349 @@
+as the current `NestedOperator`.  The flattening algorithm can be refactored out of the `NestedOperator`
+and re-purposed into the `InnerHitsOperator`. 
+
+```mermaid
+classDiagram
+    class InnerHitsOperator {
+        -PhysicalPlan input
+        -Set~String~ fields
+        -Map~String_List~String~~ groupedPathsAndFields
+        -List~Map~String_ExprValue~~ result
+        -ListIterator~Map~String_ExprValue~~ flattenedResult
+
+        +accept(PhysicalPlanNodeVisitor~R_C~ visitor, C context) ~R_C~ R
+        +getChild() List~PhysicalPlan~
+        +hasNext() boolean
+        +next() ExprValue
+        +generateFlatennedFieldsMap(ExprValue inputMap)
+        -flatten(String nestedField, ExprValue row, List~Map~String_ExprValue~~ prevList) List~Map~String_ExprValue~~
+        -containsSamePath(Map~String_ExprValue~ newMap) boolean
+        -getInnerHitFields(String field, String innerHitField, ExprValue row, List~Map~String_ExprValue~~ ret, ExprValue obj)
+    }
+    
+    InnerHitsOperator <|-- NestedOperator
+    InnerHitsOperator <|-- HasParentJoinOperator
+    InnerHitsOperator <|-- HasChildJoinOperator
+    InnerHitsOperator <|-- ByParentIdJoinOperator
+```
+
+# Annex
+
+## Annex A - Data Mapping for examples
+
+The following section includes data-mapping for data in Annex B that can be PUT into OpenSearch as an index mapping.
+
+```json
+{
+  "mappings": {
+    "properties": {
+      "house_relation": {
+        "type": "join",
+        "relations": {
+          "house": ["member", "retainer", "vassel_house"],
+          "vassel_house": "vassel"
+        }
+      },
+      "nickname": {
+        "type": "keyword"
+      },
+      "name": {
+        "properties": {
+          "firstname": {
+            "type": "text",
+            "fielddata": true
+          },
+          "lastname": {
+            "type": "text",
+            "fielddata": true
+          }
+        }
+      },
+      "gender": {
+        "type": "text",
+        "fields": {
+          "keyword": {
+            "type": "keyword"
+          }
+        }
+      },
+      "parents": {
+        "properties": {
+          "father": {
+            "type": "keyword"
+          },
+          "mother": {
+            "type": "keyword"
+          }
+        }
+      },
+      "titles": {
+        "type": "nested",
+        "properties": {
+          "title": {
+            "type": "keyword"
+          }
+        }
+      },
+      "words": {
+        "type": "text",
+        "fielddata": true
+      },
+      "house": {
+        "properties": {
+          "name": {
+            "type": "keyword"
+          },
+          "sigil": {
+            "type": "keyword"
+          },
+          "seat": {
+            "type": "keyword"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+## Annex B - Data for examples
+
+The following section includes data that has been formatted to be used by the OpenSearch `got/_bulk` ingest endpoint.
```

Comment:
Mention that all data to be added by single index named `got` (not multiple indices)